lib/internal/process/next_tick.js

```diff
@@ -32,32 +32,53 @@ function setupNextTick() {
   const kHasScheduled = 0;
   const kHasPromiseRejections = 1;
 
-  const nextTickQueue = {
-    head: null,
-    tail: null,
+  // Queue size for each tick array. Must be a factor of two.
+  const kQueueSize = 2048;
+  const kQueueMask = kQueueSize - 1;
+
+  class FixedQueue {
+    constructor() {
+      this.btm = 0;
+      this.top = 0;
+      this.list = new Array(kQueueSize);
+      this.next = null;
+    }
+
     push(data) {
-      const entry = { data, next: null };
-      if (this.tail !== null) {
-        this.tail.next = entry;
-      } else {
-        this.head = entry;
-        tickInfo[kHasScheduled] = 1;
-      }
-      this.tail = entry;
-    },
+      this.list[this.top] = data;
+      this.top = (this.top + 1) & kQueueMask;
+    }
+
     shift() {
-      if (this.head === null)
-        return;
-      const ret = this.head.data;
-      if (this.head === this.tail) {
-        this.head = this.tail = null;
+      const next = this.list[this.btm];
+      if (next === undefined) return null;
+      this.list[this.btm] = undefined;
+      this.btm = (this.btm + 1) & kQueueMask;
+      return next;
+    }
+  }
+
+  var head = new FixedQueue();
+  var tail = head;
+
+  function push(data) {
+    if (head.list[head.top] !== undefined)
+      head = head.next = new FixedQueue();
+    head.push(data);
+    if (tickInfo[kHasScheduled] === 0)
```

Comment:
Manipulating and accessing the Aliased Buffer is somewhat expensive, from my past testing, is there any other way to check this condition? I'm guessing tail.top === tail.btm or something similar might do the trick?

Reply:
Ah, ya lots of easy ways. Wasn't aware that this was expensive :)

Reply:
I mean, it's not a huge cost or anything but unless we need to specifically check the data in it, it's better to use other available conditions. Since `push` is a hot path, this could make a small difference.